process_report/tests/unit/invoices/test_invoice_fetch.py

```diff
@@ -0,0 +1,35 @@
+import pandas as pd
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from pytest_mock.plugin import MockerFixture
+
+from process_report.tests import util as test_utils
+
+
+def test_fetch_with_mock_s3_bucket(mocker: "MockerFixture") -> None:
+    """Test that fetch() out-of-band utility function loads data correctly when S3 bucket is mocked."""
+
+    test_invoice = test_utils.new_base_invoice(
+        name="TestInvoice", invoice_month="2024-08"
+    )
+    expected_data = pd.DataFrame(
+        {
+            "Invoice Month": ["2024-08", "2024-08"],
+            "Project - Allocation": ["P1", "P2"],
+            "Manager (PI)": ["pi1@bu.edu", "pi2@harvard.edu"],
+            "Cost": [100.00, 150.00],
+        }
+    )
+
+    mocker.patch("pandas.read_csv", return_value=expected_data)
+    mock_s3_bucket = mocker.Mock()
+    mock_s3_bucket.download_file.return_value = None
+
+    test_invoice.fetch(mock_s3_bucket)
+
+    assert test_invoice.data is not None
+    pd.testing.assert_frame_equal(test_invoice.data, expected_data)
```

Comment:
I would suggest 
```suggestion
    assert test_invoice.data.equals(expected_data)
```
I think this looks simpler, and is also what the rest of the repo does